### ISSUE TITLE
'Uncaught Error: vector<T> too long'

### ISSUE THREAD

**Issue by springmeyer** [mapnik/node-mapnik#406]
I just merged @BergWerkGIS's updates from master into the `release/image_any` branch. It is currently expected that the compile of master will fail against this new SDK but the `release/image_any` should compile cleanly. It looks like it does compile fine, but there are some curious new test failures that all say `Uncaught Error: vector<T> too long`:
https://ci.appveyor.com/project/Mapbox/node-mapnik/build/1.0.224/job/q6he69fhehr4dlpg#L1721.

No idea what is causing this. Maybe some change related to the image work is corrupting memory and causing tests to fail afterward?

Note: @flippmoke only focus on the node 0.11.x appveyor runs because a bug in node/mocha effectively hides the errors for the node 0.10.x runs. Also @BergWerkGIS is offline tomorrow (monday).

**Comment by springmeyer** [comment]
Looks like this error is coming out of `vtile.render`. So `err` is being populated, for example, right here: https://github.com/mapnik/node-mapnik/blob/fb201eb533b24013b26c16a772590a1ae5b23977/test/vector-tile.composite.test.js#L139. This seems to indicate that image allocation after overflow might be the cause, somewhere internal to mapnik-vector-tile.

**Comment by springmeyer** [comment]
I was able to replicate the error after building mapnik master from source and node-mapnik master on windows. So @flippmoke I don't suspect any problem in the SDK anymore: more likely a code-level bug somewhere.

**Comment by springmeyer** [comment]
spent a while debugging on windows. It turns out that neither of these operators are every hit: https://github.com/mapnik/node-mapnik/blob/master/src/mapnik_vector_tile.cpp#L2564-L2578.

I tried:

```
            agg_renderer_visitor visit(map_in, 
                                       m_req, 
                                       closure,
                                       map_proj,
                                       layers,
                                       tiledata,
                                       scale_denom);
            mapnik::image_rgba8 & im = mapnik::util::get<mapnik::image_rgba8>(*closure->im->get());
            visit(im);
```

But had the same error: neither of the operators are hit (based on print statements in each).

**Comment by springmeyer** [comment]
fixed by 43cec4b. https://ci.appveyor.com/project/Mapbox/node-mapnik/build/1.0.245. Not sure why the visitor was not working. @artemp - any ideas?
